### PR TITLE
login screen wip v3

### PR DESCRIPTION
**Pull request by dawoodkhan4050**
- login attempts
- login screen code refactor

### CODE REVIEW

File: src/store/index.js [hunk 1]
```diff
@@ -18,7 +23,8 @@ export default function configureStore() {
         auth: AuthReducer,
         settings: SettingReducer,
         carePlan: CarePlanReducer,
-        clients: ClientReducer
+        clients: ClientReducer,
+        loginAttempts: LoginAttemptsReducer
```
Comment (faheemsajjad65) [inline src/store/index.js:27]: @dawoodkhan4050 @hawaism  can we use auth for both authenticate or loginAttempts ?

File: src/services/auth.service.js [hunk 2]
```diff
@@ -1,20 +1,16 @@
...
-        });
+const login = async (username, password) => {
+    const response = await axios.post(API_URL + "login", {
+        username,
+        password,
+    })
+    if(response.data.token){
+        localStorage.setItem("user", JSON.stringify(response.data));
+    }
+    return response;
```
Comment (faheemsajjad65) [inline src/services/auth.service.js:13]: @hawaism handle edge case / error cases

Comment (faheemsajjad65) [reply src/services/auth.service.js:13]: what if api didnt get any token ?

File: src/reducers/loginAttempts.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+export default function(state = initialState, action) {
+    const { type, payload } = action;
+
+    switch (type) {
+        case LOGIN_ATTEMPT_INIT:
+            return state;
+        case LOGIN_ATTEMPT_UPDATE:
+            return {
+                ...state,
```
Comment (faheemsajjad65) [inline src/reducers/loginAttempts.js:20]: @hawaism create and named function and then export that function else ESLINT gives warnings.

File: src/components/LoginForm.js [hunk 4]
```diff
@@ -102,12 +113,14 @@ export default function LoginForm() {
...
         // console.log("submitted data ",data)
-        setLoginError(null);
-        setHasLoginPending(true);
-        dispatch(login(data.username, data.password)).then(
-            handleLoginSuccess,
-            handleLoginFail
-        );
+        if(!isRetryLoginBlocked){
+            setLoginError(null);
+            setHasLoginPending(true);
```
Comment (faheemsajjad65) [inline src/components/LoginForm.js:118]: @hawaism i dnt understand this loginPending function.
can you please put some light on it .
As creates mess for a simple login

File: src/services/auth.service.js (hunk 2 above)
Comment (dawoodkhan4050) [reply src/services/auth.service.js:13]: yes right now it is not handling any edge cases. there are some changes on API, will updating it accordingly.

File: src/reducers/loginAttempts.js (hunk 3 above)
Comment (dawoodkhan4050) [reply src/reducers/loginAttempts.js:20]: alright

File: src/components/LoginForm.js (hunk 4 above)
Comment (dawoodkhan4050) [reply src/components/LoginForm.js:118]: loginPending will disable the login button until api response.

File: src/store/index.js (hunk 1 above)
Comment (dawoodkhan4050) [reply src/store/index.js:27]: yes we can. will update accordingly

**Pull request closed without merging**